### ISSUE TITLE
Primary Category isn't Exportable

### ISSUE THREAD

**Issue by Geodyssian** [Modorganizer2/modorganizer#450]
When trying to export to csv, if I were to click on the Primary_Category, it will state "export failed: invalid index 59". Help.

# Environment

* Mod Organizer Version that exhibits the issue: 2.1.4
* Desktop OS/version used to run Mod Organizer: Windows 10

[usvfs-2018-07-25_04-46-05.log](https://github.com/Modorganizer2/modorganizer/files/2226691/usvfs-2018-07-25_04-46-05.log)

[mo_interface.log](https://github.com/Modorganizer2/modorganizer/files/2226693/mo_interface.log)

**Comment by Al12rs** [comment]
Is the primary category that important? I was thinking of actually removing the option all together from there since it doesn't seem to work

**Comment by Geodyssian** [comment]
It is when you're working with 400+ mods.

**Comment by twizz0r** [comment]
I'm getting "export failed: invalid index 86"

Category 86 is defined and in use. 

Any ideas? 

https://gist.github.com/twizz0r/6350be9bdd3c726ada17cb2490c3fe33#file-categories-dat

**Comment by Al12rs** [comment]
Does the export fail only with categories selected? Index could be of the mod instead of the category

**Comment by twizz0r** [comment]
Yes that's the case. Would i be looking for a mod that has not meta.ini or no category defined in meta.ini? 

All of my mods are categorized but i noticed a few of my merged patches have no meta.ini.

**Comment by Al12rs** [comment]
Opening MO2 again should fix the missing meta.ini, I doubt it's because a mod is missing the category but I can't be sure.
What is the exact error?

**Comment by Al12rs** [comment]
Try changing the category id of the 86 category to something else maybe

**Comment by twizz0r** [comment]
I took the opportunity to sort my categories and made sure that every mod has a category. 

Now the error occurs and index 92 and if i change the mods to a new category (59), then the index error shows up in the new category (59). 

Reinstalled the cat 92 mods...same story.

**Comment by twizz0r** [comment]
Editing the category id makes the index error occur in another category but not the category that had the id changed. Tried it three times, same behavior.

**Comment by mdavis199** [comment]
I don't have Qt libraries installed, so I can't test and submit a patch, but I think this can be fixed changing this line:
https://github.com/ModOrganizer2/modorganizer/blob/6305f6e2e93ec9c0e6fdef996713eb5be7bdd46e/src/mainwindow.cpp#L4617

to this:
```cpp
builder.setRowField("#Primary_Category", (m_CategoryFactory.categoryExists(info->getPrimaryCategory())) ? m_CategoryFactory.getCategoryNameByID(info->getPrimaryCategory()) : "");
```

The  call to `getCategoryName` should be to `getCategoryNameById`.

Also, please don't remove the primary category.  I've set up custom categories that follow a modding guide. It makes it super easy to sort and keep track when I rebuilding and customizing the guide. 

Thanks.

**Comment by Al12rs** [comment]
You are right, nice catch!
Will be in the next dev build on discord.
